ignite/pkg/jsonfile/jsonfile.go

```diff
@@ -0,0 +1,311 @@
+package jsonfile
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/buger/jsonparser"
+	"github.com/pkg/errors"
+
+	"github.com/ignite-hq/cli/ignite/pkg/tarball"
+)
+
+const (
+	keySeparator = "."
+)
+
+type (
+	// JSONFile represents the file
+	JSONFile struct {
+		file        readWriteSeeker
+		tarballPath string
+		url         string
+		updates     map[string][]byte
+		cache       []byte
+	}
+	// UpdateFileOption configures file update.
+	UpdateFileOption func(map[string][]byte)
+
+	writeTruncate interface {
+		Truncate(size int64) error
+	}
+	readWriteSeeker interface {
+		io.ReadWriteSeeker
+		Close() error
+		Sync() error
+	}
+)
+
+var (
+	// ErrFieldNotFound parameter not found into json
+	ErrFieldNotFound = errors.New("JSON field not found")
+	// ErrInvalidValueType invalid value type
+	ErrInvalidValueType = errors.New("invalid value type")
+	// ErrInvalidURL invalid file URL
+	ErrInvalidURL = errors.New("invalid file URL")
+)
+
+// New creates a new JSONFile
+func New(file readWriteSeeker) *JSONFile {
```

Comment:
```suggestion
func New(file ReadWriteSeeker) *JSONFile {
```

public func should accept public types and return public types